archinstall/lib/user_interaction/system_conf.py

```diff
@@ -125,9 +125,11 @@ def ask_for_bootloader(advanced_options: bool = False, preset: str = None) -> st
 	else:
 		preset_val = preset
 
-	bootloader = "systemd-bootctl" if has_uefi() else "grub-install"
+	bootloader = "grub-install"
 
-	if has_uefi():
+	# systemd-boot doesn't support 32-bit UEFIs, so it must use GRUB.
+	# This was already fixed upstream, but hasn't been added to systemd-stable!
```

Comment:
Well, with that, is this PR even necessary anymore? It was intended to use GRUB as a fallback in case of a 32-bit UEFI, but now that systemd-boot works the process should be identical to a 64-bit UEFI.

I forget if I made any other changes for compatibility (other than the 32-bit UEFI debug info), it's been so long!

Reply:
> Well, with that, is this PR even necessary anymore?

Yes.

The ArchWiki has been updated with the following: [1][2]

> ... 32-bit IA32 UEFI; while this is supported, it will limit the boot loader choice to GRUB.

From ArchWiki admin [nl6720](https://wiki.archlinux.org/title/User:Nl6720) on the _Installation guide_ talk page: [3]

> Yes, systemd-boot supports it, but the [systemd](https://archlinux.org/packages/?name=systemd) package does not build and package IA32 EFI binaries.

[1] https://wiki.archlinux.org/title/Installation_guide#Verify_the_boot_mode
[2] https://wiki.archlinux.org/index.php?title=Installation_guide&diff=prev&oldid=782855
[3] https://wiki.archlinux.org/title/Talk:Installation_guide#Use_/sys/firmware/efi/fw_platform_size_to_verify_the_boot_mode